crates/worktree/src/worktree.rs

```diff
@@ -3136,11 +3136,74 @@ async fn is_git_dir(path: &Path, fs: &dyn Fs) -> bool {
 async fn build_gitignore(abs_path: &Path, fs: &dyn Fs) -> Result<Gitignore> {
     let contents = fs.load(abs_path).await?;
     let parent = abs_path.parent().unwrap_or_else(|| Path::new("/"));
+
     let mut builder = GitignoreBuilder::new(parent);
     for line in contents.lines() {
         builder.add_line(Some(abs_path.into()), line)?;
     }
-    Ok(builder.build()?)
+
+    builder.build()?;
```

Comment:
We should continue adding to the original `builder`, instead of creating a new builder.